testing/ase/test_api_c_ase.cpp

```diff
@@ -0,0 +1,214 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+extern "C" {
+#include <json-c/json.h>
+#include <uuid/uuid.h>
+#include <opae/types.h>
+#include "types_int.h"
+#include "ase_common.h"
+#include "common_int.h"
+#include <pthread.h>
+}
+
+#include <opae/fpga.h>
+
+#include "gtest/gtest.h"
+#include "test_system.h"
+
+/**
+* @test       ase_common_01
+*
+* @brief      When the property's magic is invalid and libopae-ase-c is loaded:
+*             prop_check_and_lock() function should return FPGA_INVALID_PARAM
+*
+*/
+TEST(sim_sw_ase, ase_common_01) {
+	struct _fpga_properties prop;
+
+	prop.lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+	prop.magic = FPGA_PROPERTY_MAGIC;
+	EXPECT_EQ(FPGA_OK, prop_check_and_lock(&prop));
+
+	prop.magic = 0xFFFFFFFF;
+	EXPECT_EQ(FPGA_INVALID_PARAM, prop_check_and_lock(&prop));
+
+
+}
+
+/**
+* @test       ase_common_02
+*
+* @brief      When the property's magic is invalid and libopae-ase-c is loaded:
+*             prop_check_and_lock() function should return FPGA_INVALID_PARAM
+*
+*/
+TEST(sim_sw_ase, ase_common_02) {
+	struct _fpga_handle handle;
+
+	handle.magic = 0xFFFFFFFF;
+	handle.lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+	EXPECT_EQ(FPGA_INVALID_PARAM, handle_check_and_lock(&handle));
+}
+
+/**
+* @test       ase_common_03
+*
+* @brief      When the property's magic is invalid and libopae-ase-c is loaded:
+*             prop_check_and_lock() function should return FPGA_INVALID_PARAM
+*
+*/
+TEST(sim_sw_ase, ase_common_03) {
+	struct _fpga_event_handle ehandle;
+
+	ehandle.magic = 0xFFFFFFFF;
+	ehandle.lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+	EXPECT_EQ(FPGA_INVALID_PARAM, event_handle_check_and_lock(&ehandle));
+}
+
+/**
+ * @test       fpga_version
+ *
+ * @brief      When the parameter fpga_version* to fpgaGetOPAECVersion() is nullptr, 
+ *             the function returns FPGA_INVALID_PARAM.
+ *             If the parameter fpga_version* to fpgaGetOPAECVersion() is valid,
+ *             the function returns FPGA_OK.
+ *             When the parameter version_str* to fpgaGetOPAECVersionString() is nullptr, 
+ *             the function returns FPGA_INVALID_PARAM.
+ *             When the parameter build_str* to fpgaGetOPAECBuildString() is nullptr, 
+ *             the function returns FPGA_INVALID_PARAM.
+ */
+TEST(open_c_ase, fpga_version) {
+	EXPECT_EQ(FPGA_INVALID_PARAM, fpgaGetOPAECVersion(nullptr));
+	fpga_version version;
+	EXPECT_EQ(FPGA_OK, fpgaGetOPAECVersion(&version));
+
+	EXPECT_EQ(FPGA_INVALID_PARAM, fpgaGetOPAECVersionString(nullptr, 10));
+	EXPECT_EQ(FPGA_INVALID_PARAM, fpgaGetOPAECBuildString(nullptr, 10));
+}
+
+/**
+ * @test       ase_wsid_1
+ *
+ * @brief      When the parameter n_hash_buckets is zero or greater than 16384, 
+ *             the function returns NULL. 
+ */
+TEST(open_c_ase, ase_wsid_1) {
+	EXPECT_EQ(NULL, wsid_tracker_init(0));
+	EXPECT_EQ(NULL, wsid_tracker_init(16385));
```

Comment:
Is the limit implemented as a #define somewhere that we can use, ie
```c
EXPECT_EQ(NULL, wsid_tracker_init(MAX_WSID_TRACKER_BUCKETS+1));
```

Reply:
Added #define